src/coreclr/jit/importercalls.cpp

```diff
@@ -7727,11 +7727,35 @@ void Compiler::impCheckCanInline(GenTreeCall*           call,
                 return;
             }
 #endif
+            JITDUMP("\nCheckCanInline: fetching method info for inline candidate %s -- context %p\n",
+                    compiler->eeGetMethodName(ftn), pParam->exactContextHnd);
+
+            if (pParam->exactContextHnd == nullptr)
+            {
+                JITDUMP("NULL context\n");
+            }
+            else if (pParam->exactContextHnd == METHOD_BEING_COMPILED_CONTEXT())
+            {
+                JITDUMP("Current method context\n");
+            }
+            else if ((((size_t)pParam->exactContextHnd & CORINFO_CONTEXTFLAGS_MASK) == CORINFO_CONTEXTFLAGS_METHOD))
+            {
+                JITDUMP("Method context: %s\n",
+                        compiler->eeGetMethodName((CORINFO_METHOD_HANDLE)pParam->exactContextHnd));
+            }
+            else
+            {
+                JITDUMP("Class context: %s\n", compiler->eeGetClassName((CORINFO_CLASS_HANDLE)(
+                                                   (size_t)pParam->exactContextHnd & ~CORINFO_CONTEXTFLAGS_MASK)));
+            }
+
+            const bool isGdv = pParam->call->IsGuardedDevirtualizationCandidate();
 
             // Fetch method info. This may fail, if the method doesn't have IL.
+            // NOTE: For GDV we're expected to use a different context (per candidate)
             //
             CORINFO_METHOD_INFO methInfo;
-            if (!compCompHnd->getMethodInfo(ftn, &methInfo))
+            if (!compCompHnd->getMethodInfo(ftn, &methInfo, isGdv ? nullptr : pParam->exactContextHnd))
```

Comment:
Same comment as in #87847 -- change we change `impMarkInlineCandidateHelper` to pass the right context through for GDVs instead?

Reply:
I suppose that bug is that we don't have the right context saved with the GDVs, and maybe passing `nullptr` through is going to break things even further. So disregard this if that's the case.

Reply:
> I suppose that bug is that we don't have the right context saved with the GDVs, and maybe passing `nullptr` through is going to break things even further. So disregard this if that's the case.

Exactly, this is a bug I spent quite some time tryint to trace - if we can't provide the exact context we'd better provide null here. Ideally, with proper fix for https://github.com/dotnet/runtime/pull/87847 I will be able to remove `isGdv` condition here